osu.Game/Tests/Visual/Spectator/TestSpectatorClient.cs

```diff
@@ -37,6 +37,7 @@ public class TestSpectatorClient : SpectatorClient
         private readonly Dictionary<int, ReplayFrame> lastReceivedUserFrames = new Dictionary<int, ReplayFrame>();
 
         private readonly Dictionary<int, int> userBeatmapDictionary = new Dictionary<int, int>();
+        private readonly Dictionary<int, APIMod[]> userModsDictionary = new Dictionary<int, APIMod[]>();
```

Comment:
As per the test failures, the dictionary should also be filled when `BeginPlayingInternal` is called.